sui/manage-coins.js

```diff
@@ -0,0 +1,218 @@
+            }
+        } else {
+            for (const coinType in coinTypeToCoins) {
+                const coins = coinTypeToCoins[coinType];
+                if (this.isGasToken(coins.data[0])) continue;
+                const [coin] = CoinManager.doSplitCoins(tx, coins, splitAmount);
+
+                if (options.transfer) {
+                    CoinManager.doTransfer(tx, coin, options.transfer);
+                }
+            }
+        }
+
+        if (options.transfer) {
+            console.log(`\nTransfer ${splitAmount} coins for every split coin to ${chalk.green(options.transfer)}`);
+        }
+
+        // The transaction will fail if the gas budget is not set for splitting coins transaction
+        tx.setGasBudget(1e8);
+    }
+
+    static doTransfer(tx, coin, recipient) {
+        tx.transferObjects([coin], recipient);
+    }
+
+    static doSplitCoins(tx, coins, splitAmount) {
+        const firstObjectId = this.isGasToken(coins.data[0]) ? tx.gas : coins.data[0].coinObjectId;
+        const response = tx.splitCoins(firstObjectId, [splitAmount]);
+        console.log(`Split coins of type '${chalk.green(coins.data[0].coinType)}' with amount ${splitAmount}`);
+        return response;
+    }
+
+    static async mergeCoin(tx, coinTypeToCoins, options) {
+        const coinType = options.coinType;
+        console.log('\n==== Merging Coins ====');
+
+        // Throw an error if the coin type is specified but no coins are found
+        CoinManager.checkCoinType(coinType, coinTypeToCoins);
+
+        if (coinType) {
+            const coins = coinTypeToCoins[coinType];
+            await CoinManager.doMergeCoin(tx, coins);
+        } else {
+            for (const coinType in coinTypeToCoins) {
+                const coins = coinTypeToCoins[coinType];
+                await CoinManager.doMergeCoin(tx, coins);
+            }
+        }
+    }
+
+    static async doMergeCoin(tx, coins) {
+        const coinObjectIds = coins.data.map((coin) => coin.coinObjectId);
+
+        // If the first coin is a gas token, remove it from the list. Otherwise, the merge will fail.
+        if (CoinManager.isGasToken(coins.data[0])) {
+            coinObjectIds.shift();
+        }
+
+        if (coinObjectIds.length < 2) {
+            // Need at least 2 coins to merge
+            return;
+        }
+
+        const firstCoin = coinObjectIds.shift();
+        const remainingCoins = coinObjectIds.map((id) => tx.object(id));
+
+        tx.mergeCoins(firstCoin, remainingCoins);
+        console.log(`Merge ${coins.data.length} coins of type '${chalk.green(coins.data[0].coinType)}'`);
+    }
+
+    static async processCommand(config, chain, options) {
+        const [keypair, client] = getWallet(chain, options);
+        printInfo('Wallet Address', keypair.toSuiAddress());
+
+        const coinTypeToCoins = await CoinManager.getAllCoins(client, keypair.toSuiAddress());
+        CoinManager.printAllCoins(coinTypeToCoins);
```

Comment:
Looking into the Sui SDK, we could use `getAllBalances` method for printing balances or `getBalances` if we want to print it for a given coin instead of `getAllCoins`

Reply:
yes, it's exactly the same. however, we still need the data from `getAllCoins` api for merge and split mode, so i think we could just reuse the data from it to print coin's balance details.